### PR TITLE
Unequal → Not equal

### LINKED ISSUE #109168: [docs] Iterator fuction `ne` description should use "not equal"
Hi, I'm currently learning Rust and English is not my first language on top of which I suffer from dyslexia.
The language style in docs is a bit difficult for me to understand sometimes. 

For example docs for the `not equal` function feels a bit misleading by using the synonym `unequal` instead of `not equal` while even the name of the function is the abbreviation of the words `Not Equal`.

https://github.com/rust-lang/rust/blob/992d154f3a84cc8abcefcf6e6cf3698e4821b506/library/core/src/iter/traits/iterator.rs#L3724

I'd prefer if docs reflected it and said
```
/// Determines if the elements of this [`Iterator`] are not equal to those of 
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes #109168

### CODE REVIEW

File: library/std/src/sync/remutex.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ use crate::sys::locks as sys;
 /// `owner` can be checked by other threads that want to see if they already
 /// hold the lock, so needs to be atomic. If it compares equal, we're on the
 /// same thread that holds the mutex and memory access can use relaxed ordering
-/// since we're not dealing with multiple threads. If it compares unequal,
+/// since we're not dealing with multiple threads. If it's not equal
```
Comment (compiler-errors) [inline library/std/src/sync/remutex.rs:38]:
```suggestion
/// since we're not dealing with multiple threads. If it's not equal,
```

**Comment by compiler-errors** [comment]
please squash this into one commit, then i can approve

**Comment by gimbles** [comment]
(Sorry for the ping spam MIR & RA peeps, just a bunch of comment changes 😅)

**Comment by compiler-errors** [comment]
Also, actually, the guidance is typically to leave tools alone. Can you revert that rust-analyzer change?

**Comment by gimbles** [comment]
Done & Squashed 👏🏻

**Pull request merged**